pinot-clients/pinot-java-client/src/main/java/org/apache/pinot/client/JsonAsyncHttpPinotClientTransportFactory.java

```diff
@@ -85,8 +85,9 @@ public JsonAsyncHttpPinotClientTransportFactory withConnectionProperties(Propert
       _headers = ConnectionUtils.getHeadersFromProperties(properties);
     }
 
-    if (_scheme == null) {
-      _scheme = properties.getProperty("scheme", CommonConstants.HTTP_PROTOCOL);
+    String scheme = properties.getProperty("scheme", CommonConstants.HTTP_PROTOCOL);
+    if (_scheme == null || !_scheme.contentEquals(scheme)) {
+      _scheme = scheme;
     }
```

Comment:
this is equivalent to setting directly right?
```
_scheme = properties.getProperty("scheme", CommonConstants.HTTP_PROTOCOL);
```
in what scenarios is this not true?

Reply:
When someone is using Java Client in their code and do not use the builder pattern, the scheme:https property never gets enforced

Reply:
i dont understand. the code is literally setting the class variable `_schema` to the property value `schema` in any cases the schema variable is different from _schema?

Reply:
https://github.com/apache/pinot/issues/14500